itest/round_trip_send_test.go

```diff
@@ -0,0 +1,127 @@
+		t.t, t, t.lndHarness.BackendCfg, t.lndHarness.Bob, t.universeServer,
+	)
+	defer func() {
+		require.NoError(t.t, secondTarod.stop(true))
+	}()
+
+	// We'll send half of the minted units to Bob, and then have Bob return
+	// half of the units he received.
+	fullAmt := rpcAssets[0].Amount
+	bobAmt := fullAmt / 2
+	aliceAmt := bobAmt / 2
+
+	// First, we'll send half of the units to Bob.
+	bobAddr, err := secondTarod.NewAddr(ctxb, &tarorpc.NewAddrRequest{
+		GenesisBootstrapInfo: genBootstrap,
+		Amt:                  bobAmt,
+	})
+	require.NoError(t.t, err)
+
+	assertAddrCreated(t.t, secondTarod, rpcAssets[0], bobAddr)
+	sendResp := sendAssetsToAddr(t, t.tarod, bobAddr)
+	sendRespJSON, err := formatProtoJSON(sendResp)
+	require.NoError(t.t, err)
+	t.Logf("Got response from sending assets: %v", sendRespJSON)
+
+	// Mine a block to force the send we created above to confirm.
+	_ = mineBlocks(t, t.lndHarness, 1, len(rpcAssets))
+
+	// Complete the send by exhanging proofs.
+	sendProof(t, t.tarod, secondTarod, bobAddr, genInfo)
+
+	// Now, Alice will request half of the assets she sent to Bob.
+	aliceAddr, err := t.tarod.NewAddr(ctxb, &tarorpc.NewAddrRequest{
+		GenesisBootstrapInfo: genBootstrap,
+		Amt:                  aliceAmt,
+	})
+	require.NoError(t.t, err)
+
+	assertAddrCreated(t.t, t.tarod, rpcAssets[0], aliceAddr)
+	sendResp = sendAssetsToAddr(t, secondTarod, aliceAddr)
+	sendRespJSON, err = formatProtoJSON(sendResp)
+	require.NoError(t.t, err)
+	t.Logf("Got response from sending assets: %v", sendRespJSON)
+
+	// Mine a block to force the send we created above to confirm.
+	_ = mineBlocks(t, t.lndHarness, 1, len(rpcAssets))
+
+	// Complete the send by exhanging proofs.
+	sendProof(t, secondTarod, t.tarod, aliceAddr, genInfo)
+
+	// Check the final state of both nodes. Each node should list
+	// one transfer, and Alice should have 3/4 of the total units.
+	err = wait.NoError(func() error {
+		transferResp, err := t.tarod.ListTransfers(
+			ctxb, &tarorpc.ListTransfersRequest{},
+		)
+		require.NoError(t.t, err)
+		require.Len(t.t, transferResp.Transfers, 1)
+		require.Len(t.t, transferResp.Transfers[0].AssetSpendDeltas, 1)
+		delta := transferResp.Transfers[0].AssetSpendDeltas[0]
+		require.Equal(t.t, delta.NewAmt, bobAmt)
+
+		balancesResp, err := t.tarod.ListBalances(
+			ctxb, &tarorpc.ListBalancesRequest{
+				GroupBy: &tarorpc.ListBalancesRequest_AssetId{
+					AssetId: true,
+				},
+			},
+		)
+		require.NoError(t.t, err)
+		require.Len(t.t, balancesResp.AssetBalances, 1)
+		balances := maps.Values(balancesResp.AssetBalances)
+		require.Equal(t.t, balances[0].Balance, bobAmt+aliceAmt)
+
+		transferResp, err = secondTarod.ListTransfers(
+			ctxb, &tarorpc.ListTransfersRequest{},
+		)
+		require.NoError(t.t, err)
+		require.Len(t.t, transferResp.Transfers, 1)
+		require.Len(t.t, transferResp.Transfers[0].AssetSpendDeltas, 1)
+		delta = transferResp.Transfers[0].AssetSpendDeltas[0]
+		require.Equal(t.t, delta.NewAmt, aliceAmt)
```

Comment:
Can lift this into a helper assert func, params here are the amt, and the node to issue the query against. Same for the balances call.